### PR TITLE
Fixed the default value of framework.form.enabled

### LINKED ISSUE #8412: Wrong default value for "framework.form.enabled" config node
As of Symfony 3.3 the `framework.form.enabled` config option is default ~`true`~  `false` [if you are requiring `symfony/symfony`](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L120), but documentation says it is always false by default (https://symfony.com/doc/current/reference/configuration/framework.html#reference-form-enabled).

Originally reported on https://github.com/symfony/symfony/issues/24269
Also applies to assets, validation, translation, serialization and csrf

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #8412.

### CODE REVIEW

**Review by xabbuh — APPROVED** [review]

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -389,12 +389,15 @@ form
...
 
-**type**: ``boolean`` **default**: ``false``
+**type**: ``boolean`` **default**: (see explanation below)
 
 Whether to enable the form services or not in the service container. If
 you don't use forms, setting this to ``false`` may increase your application's
 performance because less services will be loaded into the container.
 
+The default value is ``true`` when the project uses ``symfony/symfony`` as a
+dependency. Otherwise, it defaults to ``false``.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: This is incorrect. The default is still disabled. The only case where it is automatically enabled is when you're using `symfony/flex` *and* `symfony/form` is required as well.

This also applies to assets, validation, translation, serialization and csrf. Can we add the note there as well?

**Review by wouterj — CHANGES_REQUESTED** [review]

File: reference/configuration/framework.rst (hunk 1 above)
Comment (chalasr) [reply reference/configuration/framework.rst]: Indeed, issue updated accordingly

Comment (weaverryan) [reply reference/configuration/framework.rst]: Hmm. So I think explaining the logic behind the default value is too complex and doesn't add much value. A more useful thing is to run `debug:config framework`. I think we should mention this command at the top of each reference, and remove the `default` value in cases where there is not a simple answer.

**Comment by javiereguiluz** [comment]
Closed in favor of #8979.

**Pull request closed without merging**